website/src/content/api-reference/commercetools-frontend-application-shell.mdx

```diff
@@ -0,0 +1,327 @@
+---
+title: "@/application-shell"
+---
+
+<Subtitle>
+
+React component that ties together all the main functionalities of the Custom Applications.
+
+</Subtitle>
+
+# Installation
+
+```
+yarn add @commercetools-frontend/application-shell
+
+# or
+
+npm --save install @commercetools-frontend/application-shell
+```
+
+Additionally install the peer dependencies (if not present)
+
+```
+yarn add @apollo/client react react-dom react-intl react-redux react-router-dom redux @testing-library/react @testing-library/react-hooks
+
+# or
+
+npm --save install @apollo/client react react-dom react-intl react-redux react-router-dom redux @testing-library/react @testing-library/react-hooks
+```
+
+# Components
+
+## ApplicationShell
+
+This is the main component that contains all the general logic to render a Custom Application.<br />
+The `ApplicationShell` component is expected to be rendered as the top-level component your application.
+
+### Usage
+
+```jsx title="entry-point.js"
+import { ApplicationShell } from '@commercetools-frontend/application-shell';
+
+const loadMessages = (locale) => {
+  // ...
+}
+const entryPointUriPath = 'avengers'
+
+const AsyncApplicationRoutes = React.lazy(
+  () => import('../../routes' /* webpackChunkName: "avengers" */)
+);
+
+const EntryPoint = () => (
+  <ApplicationShell
+    environment={window.app}
+    applicationMessages={loadMessages}
+    onRegisterErrorListeners={() => {/* noop */}}
+  >
+    <AsyncApplicationRoutes />
+  </ApplicationShell>
+);
+export default EntryPoint;
+```
+
+### Properties
+
+| Props                            | Type               |        Required         | Default | Description                                                                                                                                                                                                                                                                                                                                                                      |
```

Comment:
I think there is still the `onRegisterErrorListeners` property missing in the props table.

Reply:
Yes, I thought about it and I prefer to leave it out as it's usually not useful/necessary. It might also be that we will drop it in the future.